### ISSUE TITLE
Apresentação do Coletivo Gloma

### ISSUE THREAD

**Issue by thiagoblauthferreira** [thiagoblauthferreira/ColetivoGloma#1]

**Comment by thiagoblauthferreira** [comment]
**1. Introdução**

Nome da Plataforma: Coletivo Gloma
Propósito: Conectar doadores e voluntários para promover doações e recursos a vítimas e animais de catástrofes climáticas.
Visão Geral: A plataforma tem como objetivo facilitar a logística de coleta e distribuição de doações feitas por voluntários. Além de comunicar e gerenciar o estoque de demandas de doações para cada abrigo ou ponto de coleta.

**2. Problema**

- Necessidade de doações: Acompanhando diversos grupos de voluntários em diversas plataformas sociais foi identificado que a falta de um controle de estoque e logística afeta a coleta e distribuição das doações, sobrecarregando pontos de distribuição com excedentes de doações enquanto outros pontos demandam doações.
- Voluntariado: Os voluntários não conseguem sinalizar o volume de doações que irão fazer, além de muitos não terem como entregar as doações necessitando de coleta da mesma. Assim como voluntários que possuem veículos que podem ajudar na coleta e não sabem quem necessita de coleta.
- Conectividade: A falta de uma plataforma que conecta demandas com voluntários e logística dificulta o controle de estoque de doações e com isso gera atraso e desperdício de recursos.

**3. Solução**

- **Cadastro de doadores e voluntários:** Facilitando a criação de perfis.
- **Listagem de Necessidades:** Organizações podem listar suas necessidades de doações e voluntários.
- **Cadastro de abrigos ou pontos de distribuição:** Cadastro com informações gerais;
- **Sistema de busca:** Permitir que doadores e voluntários encontrem oportunidades baseadas em localização, tipo de causa, etc.
- **Gestão logística:** Possibilidade de informar data e horário para coleta da doação. Alerta para voluntários do bairro realizarem a coleta e entrega.
- **Controle de estoque do abrigo ou ponto de distribuição:** Conforme usuários doadores informam a quantidade de itens que serão doados da lista de necessidades dos abrigos e pontos de distribuição a plataforma atualiza o total faltante necessário.

**Histórias de Usuário**

**Cadastro de Doadores e Voluntários**
Como um usuário (doador ou voluntário), quero criar um perfil na plataforma para que eu possa participar ativamente das oportunidades disponíveis.

**Critérios de Aceitação:**

- O usuário deve poder registrar um perfil com informações básicas como nome, e-mail, senha, endereço, telefone e data de nascimento.
- O sistema deve enviar um e-mail de confirmação para validar o registro.
- O usuário deve poder escolher se deseja ser doador, voluntário, ou ambos.
- O sistema deve oferecer um formulário claro e intuitivo para a inserção de informações.

—-------------------------------------------------------------------------------------------------------------

**Listagem de Necessidades**
Como uma organização, quero listar as necessidades de doações e voluntários para que possamos receber o suporte necessário.

**Critérios de Aceitação:**

- A organização deve poder cadastrar necessidades específicas, como tipos de doações (alimentos, roupas, etc.) e tipos de voluntariado (horas de trabalho, habilidades específicas, etc.).
- Cada necessidade deve incluir detalhes como quantidade, data limite, e descrição.
- O sistema deve permitir a atualização ou remoção das necessidades listadas.

—-------------------------------------------------------------------------------------------------------------

**Cadastro de Abrigos ou Pontos de Distribuição**
Como coordenador quero cadastrar abrigos ou pontos de distribuição com informações gerais para que eles possam ser reconhecidos e acessados por doadores e voluntários.

**Critérios de Aceitação:**

- O sistema deve permitir o cadastro e edição de abrigos e pontos de distribuição com informações como nome, endereço, telefone, e pessoa de contato.
- A plataforma deve validar e aprovar os cadastros antes de torná-los públicos. (validar email/ telefone obrigatório envio de código sms/ validador de documentos/ validar se já existe este local cadastrado)
- Cada cadastro deve incluir uma descrição do abrigo ou ponto de distribuição e suas necessidades específicas.


—-------------------------------------------------------------------------------------------------------------

**Sistema de Busca**
Como um usuário (doador ou voluntário), quero buscar oportunidades com base na localização, tipo de causa, e outras preferências para que eu possa contribuir de forma eficaz.

**Critérios de Aceitação:**

- O sistema deve permitir a busca por localização (cidade, bairro, etc.), tipo de causa (educação, saúde, animal, etc.), e tipo de necessidade (doação ou voluntariado).
- A busca deve retornar resultados relevantes e bem organizados.
- Usuários devem poder filtrar e classificar os resultados de acordo com diferentes critérios, como urgência e proximidade.

—-------------------------------------------------------------------------------------------------------------

**Gestão Logística**
Como um doador, quero informar a data e horário disponíveis para a coleta da minha doação para que os voluntários possam coordenar a coleta.

Como um voluntário, quero receber alertas sobre doações a serem coletadas no meu bairro para que eu possa planejar a coleta e entrega de maneira eficiente.

**Critérios de Aceitação:**

- O sistema deve permitir que doadores informem data e horário disponíveis para coleta durante o processo de doação.
- Voluntários devem receber alertas sobre novas doações disponíveis para coleta em sua área (email ou sms).
- O sistema deve permitir o agendamento e confirmação de coletas, notificando todas as partes envolvidas (doador, voluntário e organização por email ou sms).
- O sistema deve gerar um código validador de coleta para doador e voluntário que fará a coleta.

—-------------------------------------------------------------------------------------------------------------

**Controle de Estoque do Abrigo ou Ponto de Distribuição**
Como administrador de um abrigo ou ponto de distribuição, quero que a plataforma atualize automaticamente o controle de estoque com base nas doações prometidas para que possamos saber a quantidade de itens ainda necessária.

**Critérios de Aceitação:**

- O sistema deve atualizar automaticamente o estoque do abrigo ou ponto de distribuição quando um doador se comprometer a doar itens específicos.
- A plataforma deve mostrar a quantidade total de itens doados e o número ainda necessário.
- Administradores (os 3 coordenadores cadastrados) devem poder visualizar relatórios de estoque e fazer ajustes manuais quando necessário.

**Comment by ViniciusG1403** [comment]
Olá, 
Para o backend ja tem linguagem definida? Se não, posso recomendar Java com Quarkus? é mais robusto e extremamente rápido, ótimo para a solução.

**Comment by ccaioadriano** [comment]
Boa tarde, como sugestão eu acretido que possa ser quebrar a aplicação em duas partes como por exemplo: 
- API: Java com Quarkus;
- Frontend: Angular;
Temos que pensar que a necessidade é uma aplicação multiplataforma que tenha um exelente desempenho tanto no ambiente web como no mobile.

**Comment by henrique307** [comment]
Bom dia! criei um servidor no Discord pra facilitar nossa comunicação durante o projeto, eu pessoalmente não tenho muito domínio da ferramenta então é só o básico mas já ajuda na nossa comunicação: https://discord.gg/uCDNmkYz

**Comment by ViniciusG1403** [comment]
> Boa tarde, como sugestão eu acretido que possa ser quebrar a aplicação em duas partes como por exemplo:
> 
> * API: Java com Quarkus;
> * Frontend: Angular;
>   Temos que pensar que a necessidade é uma aplicação multiplataforma que tenha um exelente desempenho tanto no ambiente web como no mobile.

Sim, o Angular casa muito bem com Java, creio que seja uma solução muito boa pra ser utilizada.



> Bom dia! criei um servidor no Discord pra facilitar nossa comunicação durante o projeto, eu pessoalmente não tenho muito domínio da ferramenta então é só o básico mas já ajuda na nossa comunicação: https://discord.gg/uCDNmkYz

Beleza, chegando em casa entro no servidor, muito obrigado!

**Comment by thiagoblauthferreira** [comment]
Olá pessoal, muito grato pelo envolvimento de vocês no projeto. Como comentei anteriormente minha capacitação em desenvolvimento é muito iniciante. Então acho super válido essa discussão entre o grupo em relação a definições de linguagens e demais tecnologias. Vou procurar manter uma ponte entre o grupo e a realidade dos voluntários e doações para mediar e orientar vocês da melhor forma possível.

Preciso de orientações de vocês em relação a necessidades de servidor web. Estou em um conversa avançada com a Locaweb para eles cederem recursos. Então por favor me digam o que vocês precisariam nesse sentido que já vou avançando com eles nisso.

**Comment by thiagoblauthferreira** [comment]
Pessoal, algum gestor de projetos no grupo? Precisamos organizar e coordenar os esforços.

**Comment by jonasaacampos** [comment]
> Pessoal, algum gestor de projetos no grupo? Precisamos organizar e coordenar os esforços.

À disposição.

**Comment by thiagoblauthferreira** [comment]
@jonasaacampos Legal, ótimo! O grupo esta deliberando tarefas e demais esforços no servidor do discord. Consegue participar lá e nos ajudar com isso? Super valeu pelo retorno.

**Comment by henrique307** [comment]
Pessoal percebi que meu invite do discord ali de cima expirou... vou deixar esse permanente aqui https://discord.com/invite/FARNSbkZKt
